app/src/main/kotlin/com/wire/android/ui/main/message/Message.kt

```diff
@@ -0,0 +1,340 @@
+                userName = "Mateusz Pachulski",
+                membership = Membership.External,
+                isLegalHold = false,
+                time = "12.23pm",
+                messageStatus = MessageStatus.Edited
+            ),
+            messageContent = MessageContent.ImageMessage("someUrl"),
+        ),
+        Message(
+            user = User("", AvailabilityStatus.Available),
+            messageHeader = MessageHeader(
+                userName = "Mateusz Pachulski",
+                membership = Membership.External,
+                isLegalHold = false,
+                time = "12.23pm",
+                messageStatus = MessageStatus.Edited
+            ),
+            messageContent = MessageContent.TextMessage(
+                messageBody = MessageBody(
+                    "This is some test message that is very very" +
+                            "very very very very" +
+                            " very very very" +
+                            "very very very very very long" +
+                            "This is some test message that is very very" +
+                            "very very very very" +
+                            " very very very" +
+                            "very very very very very long" +
+                            "This is some test message that is very very" +
+                            "very very very very" +
+                            " very very very" +
+                            "very very very very very long" +
+                            "This is some test message that is very very" +
+                            "very very very very" +
+                            " very very very" +
+                            "very very very very very long" +
+                            "This is some test message that is very very" +
+                            "very very very very" +
+                            " very very very" +
+                            "very very very very very long"
+                )
+            ),
+        ),
+    )
+
+    Message(
+        mockMessages
+    )
+}
+
+data class MessageHeader(
+    val userName: String,
+    val membership: Membership,
+    val isLegalHold: Boolean,
+    val time: String,
+    val messageStatus: MessageStatus
+)
+
+enum class MessageStatus(val stringResourceId: Int) {
+    Untouched(-1), Deleted(R.string.label_message_status_deleted), Edited(R.string.label_message_status_edited)
+}
+
+data class Message(
+    val user: User,
+    val messageHeader: MessageHeader,
+    val messageContent: MessageContent,
+) {
+    val isDeleted = messageHeader.messageStatus == MessageStatus.Deleted
+}
+
+sealed class MessageContent {
+    data class TextMessage(val messageBody: MessageBody) : MessageContent()
+    data class ImageMessage(val imageUrl: String) : MessageContent()
+}
+
+data class MessageBody(
+    val message: String
+)
+
+data class User(
+    val avatarUrl: String = "",
+    val availabilityStatus: AvailabilityStatus,
+)
```

Comment:
All these data classes should not be living here under the app/ui path as otherwise this file will become yuuuge. Can we move these to a mocked data package or something like that?